tavern/schemas/tests.schema.yaml

```diff
@@ -248,6 +249,11 @@ mapping:
                   - type: str
                   - type: int
                   - type: float
+                  - type: seq
```

Comment:
What is the rationale behind limiting the type for the parameter values? If tavern allowed here, we would be able to do parametrisation of multiple parameters by simply passing objects:

```yaml
test_name: Make sure server doubles number properly

marks:
  - parametrize:
      key: params
      vals:
        - input: 5
          output: 10
        - input: 6
          output: 12

stages:
  - name: Make sure number is returned correctly
    request:
      url: http://localhost:5000/double
      json:
        number: !int "{params.input}"
      method: POST
      headers:
        content-type: application/json
    response:
      status_code: 200
      body:
        double: !int "{params.output}"
```


```diff
diff --git a/tavern/schemas/tests.schema.yaml b/tavern/schemas/tests.schema.yaml
index 02e4ea1..9e7ffa5 100644
--- a/tavern/schemas/tests.schema.yaml
+++ b/tavern/schemas/tests.schema.yaml
@@ -245,9 +245,7 @@ mapping:
                 type: seq
                 required: true
                 sequence:
-                  - type: str
-                  - type: int
-                  - type: float
+                  - type: any
 
   _xfail:
     type: str
```

Reply:
Mainly because I would rather keep the amount of validation we have to do to a minimum. If we allowed that, we would still need to still support the old way of passing parametrized variables as well as 
- manually validating the structure of any items in `values` so that people can't pass some arbitrarily complicated structure and wonder why it isn't working
- validating that all items in the `values` have the same keys 
- validating that all items have the same type
Using `any` in the schema is something that I only ever want to do if there is absolutely no other way around the problem, because it introduces a huge amount of extra validation/testing.

If #191 is ever implemented then we can revisit this question, but for now I would rather keep it as simple as possible.